File: Python/Recursion-1/factorial.py

# given n of 1 or more, return the factorial of n, which is n * (n-1) * (n-2) ... 1
def factorial(n):
    if n == 1:
        return 1

    return n * factorial(n-1)

print(factorial(1))
print(factorial(2))
print(factorial(3))

File: JavaScript/Recursion-1/factorial.js

// given n of 1 or more, return the factorial of n, which is n * (n-1) * (n-2) ... 1
// compute the result recursively
const factorial = (n) => {
    if (n === 1) {
        return 1;
    }

    return n * factorial(n - 1);
};

console.log(factorial(1));
console.log(factorial(2));
console.log(factorial(3));
